include/hipSYCL/algorithms/algorithm.hpp

```diff
@@ -543,6 +543,99 @@ sycl::event none_of(sycl::queue &q,
   });
 }
 
+template <class ForwardIt1, class ForwardIt2>
+sycl::event equal(sycl::queue &q,
+                   ForwardIt1 first1, ForwardIt1 last1, ForwardIt2 first2,
+                    detail::early_exit_flag_t* out, const std::vector<sycl::event>& deps= {}) {
+  std::size_t problem_size = std::distance(first1, last1);
+  if(problem_size == 0)
+    return sycl::event{};
+  auto evt = detail ::early_exit_for_each(q, problem_size, out,
+                                          [=](sycl::id<1> idx) -> bool {
+                                            auto it1 = first1;
+                                            auto it2 = first2;
+                                            std::advance(it1, idx[0]);
+                                            std::advance(it2, idx[0]);
+                                            return !(*it1 == *it2);
+                                          }, deps);
+  // use memcmp to short-circuit sequential implementation
```

Comment:
I don't understand this comment. We don't use `memcmp` here?

Reply:
My comment here is wrong, but it's also not needed so I'll remove it. The comment was to note what was mentioned on cppreference:

> Sequential std::equal is not guaranteed to be short-circuit. E.g. if the first pair elements of both ranges do not compare equal, the rest of elements may also be compared. Non-short-circuit comparison may happen when the ranges are compared with [std::memcmp](https://en.cppreference.com/w/cpp/string/byte/memcmp) or implementation-specific vectorized algorithms.